website_seo_redirection/models/website_seo_redirection.py

```diff
@@ -0,0 +1,104 @@
+# -*- coding: utf-8 -*-
+# © 2016 Jairo Llopis <jairo.llopis@tecnativa.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from openerp import _, api, fields, models
+from openerp.http import request, local_redirect
+from openerp.exceptions import ValidationError
+from ..exceptions import NoRedirectionError
+
+
+class WebsiteSeoRedirection(models.Model):
+    _name = "website.seo.redirection"
+    _rec_name = "destination"
+    _order = "destination"
+    _description = "SEO controller redirections"
+    _sql_constraints = [
+        ("origin_unique", "UNIQUE(origin)", "Duplicated original URL"),
+        ("destination_unique", "UNIQUE(destination)", "Duplicated new URL"),
+    ]
+
+    origin = fields.Char(
+        string="Original URL",
+        required=True,
+        index=True,
+        help="Path where the original controller was found.",
+    )
+    destination = fields.Char(
+        string="Redirected URL",
+        required=True,
+        index=True,
+        help="Path where the controller will be found now.",
+    )
+
+    @api.multi
+    @api.constrains("origin")
+    def _check_origin(self):
+        self._url_format_check("origin")
+
+    @api.multi
+    @api.constrains("destination")
+    def _check_destination(self):
+        self._url_format_check("destination")
+
+    @api.multi
+    def _url_format_check(self, field_name):
+        display = self._fields[field_name].string
+        for s in self:
+            value = s[field_name]
+            if not value.startswith("/"):
+                raise ValidationError(_("%s must start with `/`") % display)
+            for char in "?&=#":
```

Comment:
I think this is not restrictive enough: http://www.netregister.biz/faqit.htm#1
You might want to have a look at this: http://stackoverflow.com/questions/2894902/check-for-a-valid-domain-name-in-a-string